tools/registry-list.c

```diff
@@ -136,76 +136,85 @@ main(int argc, char **argv)
         const char *variant = rxkb_layout_get_variant(l);
         const char *brief = rxkb_layout_get_brief(l);
-        bool first;
 
-        printf("- %s%s%s%s:%s:%s",
+        printf("- layout: %s\n"
+               "  variant: %s\n"
+               "  brief: %s\n"
+               "  description: %s\n",
                rxkb_layout_get_name(l),
-               variant ? "(" : "",
-               variant ? variant : "",
-               variant ? ")" : "",
-               brief ? brief : "",
+               variant ? variant : "''",
+               brief ? brief : "''",
                rxkb_layout_get_description(l));
 
+        printf("  iso639: [");
         iso639 = rxkb_layout_get_iso639_first(l);
-        if (iso639)
-            printf(":iso639-");
-        first = true;
-        while (iso639) {
-            printf("%s%s", first ? "" : ",", rxkb_iso639_code_get_code(iso639));
-            iso639 = rxkb_iso639_code_next(iso639);
-            first = false;
+        if (iso639) {
+            const char *sep = "";
+            while (iso639) {
+                printf("%s%s", sep, rxkb_iso639_code_get_code(iso639));
+                iso639 = rxkb_iso639_code_next(iso639);
+                sep = ", ";
+            }
         }
+        printf("]\n");
+        printf("  iso3166: [");
         iso3166 = rxkb_layout_get_iso3166_first(l);
-        if (iso3166)
-            printf(":iso3166-");
-        first = true;
-        while (iso3166) {
-            printf("%s%s", first ? "" : ",", rxkb_iso3166_code_get_code(iso3166));
-            iso3166 = rxkb_iso3166_code_next(iso3166);
-            first = false;
+        if (iso3166) {
+            const char *sep = "";
+            while (iso3166) {
+                printf("%s%s", sep, rxkb_iso3166_code_get_code(iso3166));
+                iso3166 = rxkb_iso3166_code_next(iso3166);
+                sep = ", ";
+            }
         }
-
-        printf("\n");
+        printf("]\n");
         l = rxkb_layout_next(l);
     }
     printf("\n");
-    printf("Options:\n");
+    printf("option_groups:\n");
     g = rxkb_option_group_first(ctx);
     assert(g); /* Empty option goups list is usually a bug or a bad xml file */
     while (g) {
         struct rxkb_option *o;
 
-        printf("- %s:%s (%s)\n",
+        printf("- name: %s\n"
+               "  description: %s\n"
+               "  allows_multiple: %s\n"
+               "  options:\n",
                rxkb_option_group_get_name(g),
                rxkb_option_group_get_description(g),
-               rxkb_option_group_allows_multiple(g) ? "multiple" : "single");
+               rxkb_option_group_allows_multiple(g) ? "true" : "false");
 
         o = rxkb_option_first(g);
         assert(o); /* Empty option list is usually a bug or a bad xml file */
         while (o) {
             const char *brief = rxkb_option_get_brief(o);
 
-            printf("  - %s:%s:%s\n",
+            printf("  - name: '%s'\n"
```

Comment:
Here the values are quoted, but above they aren't. Is this intentional?

Reply:
yes, it is. Option names contain colons and without quotes that confuses YAML parsers, e.g. the unquoted yaml like this causes a complaint because it thinks it's a badly nested dictionary.

```yaml
 options:
  - name: esperanto:qwerty
    brief: ''
    description: 'At the corresponding key in a QWERTY layout'
```

OTOH we cannot use quotes for models and layouts because some of them include single quotes and we'd have to escape those - too much effort for what is a fairly tightly controlled set of entries anyway.